src/Build/Instance/TaskRegistry.cs

```diff
@@ -660,7 +681,22 @@ private void RegisterTask(string taskName, AssemblyLoadInfo assemblyLoadInfo, st
                 _taskRegistrations[taskIdentity] = registeredTaskEntries;
             }
 
-            registeredTaskEntries.Add(new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord));
+            RegisteredTaskRecord newRecord = new RegisteredTaskRecord(taskName, assemblyLoadInfo, taskFactory, taskFactoryParameters, inlineTaskRecord);
+
+            if (overrideTask)
+            {
+                if (overriddenTasks.ContainsKey(taskName))
+                {
+                    loggingService.LogWarning(context, null, new BuildEventFileInfo(projectUsingTaskInXml.OverrideLocation), "DuplicateOverrideUsingTaskElement", taskName);
```

Comment:
I'm split but leaning slightly towards that you should always override it, so essentially moving overridenTasks[taskName] = newRecord; out of this if/else part. Warning/message difference should still exist though.

Reply:
That would make it last-one-wins, where normal behavior for UsingTasks is first-one-wins, so I think I prefer the current approach.

Reply:
I might argue that we don't have a default yet for overriding usingtasks, and this is more similar to how we do things in other parts of MSBuild, but I'm fine with this.